src/main/java/com/nimbusds/jose/crypto/impl/ECDH1PU.java

```diff
@@ -0,0 +1,306 @@
+     * @return The algorithm mode.
+     *
+     * @throws JOSEException If the JWE algorithm is not supported.
+     */
+    public static ECDH.AlgorithmMode resolveAlgorithmMode(final JWEAlgorithm alg)
+        throws JOSEException {
+
+        if (alg.equals(JWEAlgorithm.ECDH_1PU)) {
+
+            return ECDH.AlgorithmMode.DIRECT;
+
+        } else if (alg.equals(JWEAlgorithm.ECDH_1PU_A128KW) ||
+                alg.equals(JWEAlgorithm.ECDH_1PU_A192KW) ||
+                alg.equals(JWEAlgorithm.ECDH_1PU_A256KW)
+        ) {
+
+            return ECDH.AlgorithmMode.KW;
+        } else {
+
+            throw new JOSEException(AlgorithmSupportMessage.unsupportedJWEAlgorithm(
+                alg,
+                ECDHCryptoProvider.SUPPORTED_ALGORITHMS));
+        }
+    }
+
+
+    /**
+     * Returns the bit length of the shared key (derived via concat KDF)
+     * for the specified JWE ECDH algorithm.
+     *
+     * @param alg The JWE ECDH algorithm. Must be supported and not
+     *            {@code null}.
+     * @param enc The encryption method. Must be supported} and not
+     *            {@code null}.
+     *
+     * @return The bit length of the shared key.
+     *
+     * @throws JOSEException If the JWE algorithm or encryption method is
+     *                       not supported.
+     */
+    public static int sharedKeyLength(final JWEAlgorithm alg, final EncryptionMethod enc)
+        throws JOSEException {
+
+        if (alg.equals(JWEAlgorithm.ECDH_1PU)) {
+
+            int length = enc.cekBitLength();
+
+            if (length == 0) {
+                throw new JOSEException("Unsupported JWE encryption method " + enc);
+            }
+
+            return length;
+
+        } else if (alg.equals(JWEAlgorithm.ECDH_1PU_A128KW)) {
+            return 128;
+        } else if (alg.equals(JWEAlgorithm.ECDH_1PU_A192KW)) {
+            return  192;
+        } else if (alg.equals(JWEAlgorithm.ECDH_1PU_A256KW)) {
+            return  256;
+        } else {
+            throw new JOSEException(AlgorithmSupportMessage.unsupportedJWEAlgorithm(
+                alg, ECDHCryptoProvider.SUPPORTED_ALGORITHMS));
+        }
+    }
+
+    /**
+     * Derives a shared key (via concat KDF).
+     *
+     * The method should only be called in the {@link ECDH.AlgorithmMode#DIRECT} mode.
+     *
+     * The method derives the Content Encryption Key (CEK) for the "enc" algorithm,
+     * in the {@link ECDH.AlgorithmMode#DIRECT} mode.
+     *
+     * The method does not take the auth tag because the auth tag
+     * will be generated using a CEK derived as an output of this method.
+     *
+     * @param header    The JWE header. Its algorithm and encryption method
+     *                  must be supported. Must not be {@code null}.
+     * @param Z         The derived shared secret ('Z'). Must not be
+     *                  {@code null}.
+     * @param concatKDF The concat KDF. Must be initialised and not
+     *                  {@code null}.
```

Comment:
Must not be null --> why no such checks to throw an exception if so, then?

Reply:
added null checks